File: frontend/src/types/dre.ts
export type Marketplace = "todos" | "Mercado Livre" | "Shopee";

export interface Competencia {
  ano: number;
  mes: number;
}

export interface GrupoDespesa {
  itens: Record<string, string>;
  total: string;
}

export interface Dre {
  competencia: Competencia;
  marketplace: string;
  unidades: string;
  receitas: {
    mercado_livre: string;
    shopee: string;
    receita_bruta: string;
    deducoes: GrupoDespesa;
    receita_liquida: string;
  };
  cmv: string;
  lucro_bruto: string;
  despesas_operacionais: {
    mercado_livre: GrupoDespesa;
    shopee: GrupoDespesa;
    total: string;
  };
  lucro_operacional: string;
  despesas_gerais: GrupoDespesa;
  lucro_liquido: string;
  ebitda: string;
  margens: {
    bruta: string;
    operacional: string;
    liquida: string;
  };
}

// grupo -> { categoria: valor }
export type DespesasAgrupadas = Record<string, Record<string, string>>;

export interface DespesaUpsert {
  ano: number;
  mes: number;
  grupo: string;
  categoria: string;
  valor: string;
}

File: backend/app/schemas/dre.py
"""Schemas Pydantic da DRE."""
from __future__ import annotations

from decimal import Decimal

from pydantic import BaseModel, Field


class DespesaUpsert(BaseModel):
    ano: int = Field(..., ge=2000, le=2100)
    mes: int = Field(..., ge=1, le=12)
    grupo: str = Field(..., max_length=30)
    categoria: str = Field(..., max_length=80)
    valor: Decimal = Field(default=Decimal("0"))


class DespesaOut(BaseModel):
    ano: int
    mes: int
    grupo: str
    categoria: str
    valor: Decimal
